common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -108,6 +118,14 @@ protected Recycler(int maxCapacity) {
     }
 
     protected Recycler(int maxCapacity, int maxSharedCapacityFactor) {
+        this(maxCapacity, maxSharedCapacityFactor, RATIO);
+    }
+
+    protected Recycler(int maxCapacity, int maxSharedCapacityFactor, int ratio) {
+        if (ratio > 0x40000000) {
```

Comment:
this issue has come up a few times and also came up in JCTools. Folks assume `findNextPositivePowerOfTwo` does the bounds checking and could get into trouble because of this. We are also leaking the bounds which could be encapsulated in a `findNextPositivePowerOfTwo<with bounds>` check method so that all the callers must be aware of the bounds which may lead to code duplication. Can you file an issue if you don't want to take care in this PR and I'll do this as a followup?